### DISCUSSION TITLE
unwanted Factory Reset on ESP8266

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#22637] [General]
I have an issue with an ESP-12 running on 14.4.0 Something is triggering a factory reset - device comes up with default config and Wifi-Manager enabled. Tested on several ESP-12 - everywhere the same

```
10:16:17.922 RSL: POWER = OFF
10:16:18.952 RSL: RESULT = {"POWER":"ON"}
10:16:18.954 RSL: POWER = ON
10:16:19.781 RSL: RESULT = {"POWER":"OFF"}
10:16:19.783 RSL: POWER = OFF
10:16:20.682 RSL: RESULT = {"POWER":"ON"}
10:16:20.684 RSL: POWER = ON
10:16:21.388 RSL: RESULT = {"POWER":"OFF"}
10:16:21.390 RSL: POWER = OFF
10:16:22.289 RSL: RESULT = {"POWER":"ON"}
10:16:22.291 RSL: POWER = ON
10:16:24.931 RSL: RESULT = {"POWER":"OFF"}
10:16:24.934 RSL: POWER = OFF
10:16:25.748 RSL: RESULT = {"POWER":"ON"}
10:16:25.750 RSL: POWER = ON
10:17:06.328 RSL: RESULT = {"Reset":"Reset and Restarting"}
10:17:06.525 CFG: Use defaults
10:17:08.518 APP: Restarting

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.002 HDW: ESP8266EX
00:00:00.027 CFG: Loaded from flash at FA, Count 2
00:00:00.032 QPC: Count 1
00:00:00.044 Project tasmota - Tasmota Version 14.4.0(release-tasmota)-2_7_8(2024-12-11T09:18:54)
00:00:00.297 WIF: WifiManager active for 3 minutes
00:00:00.805 HTP: Web server active on tasmota-0D16E6-5862 with IP address 192.168.4.1
00:00:06.986 QPC: Reset

```
Does anybody have an idea?

**Comment by Jason2866** [comment]
The log shows power on off cycles. So the device recovery routine from Tasmota kicks in and resets the device.

**Reply by sfromis** [reply]
Since there is a bit over 40 seconds from the power on/off actions, the reason for the factory reset does not appear to be the cycling, but instead a `Button` input registering as "pushed" for 40 seconds straight.

This is generally a misconfiguration (unless the hardware does not work as it should), where a `Button` input is registering as "pushed" while you did not. Fixing that depends on what the circuit actually does, but will often be something like configuring the gpio for inverted `Button_i` instead of just `Button`.

Note: Do *not* config a Button input unless you actually have that wired up.

**Reply by tsunfinger** [reply]
correct, I using Button_i as Button only reacts when release the button. When using Button I lose 300-500ms and people moaning that button not reacting in time. Is it this forcing the reset?

**Reply by sfromis** [reply]
Yes, configuring the Button to be "pushed" all the time, except when you push it, is activating a recovery feature of a factory reset after 40 seconds of Button registering as pushed.

Wanting quick reaction for pushing a Button does not make it "right" to do a bad config. With the *correct* gpio config of non-inverted `Button`, you can avoid the latency by using `SetOption13` for "immediate action on a single button press".

The reason for the latency is multi-press actions actions for buttons, only detectable after waiting for next push, but you can disable that feature, hence you lose some options.
